util/ipgen/renderer.py

```diff
@@ -123,7 +123,10 @@ def _tplfunc_instance_vlnv(self, template_vlnv_str: str) -> str:
         # For example, a core name `rv_plic_component` will result in an
         # instance name 'my_instance_component' (instead of
         # 'my_instance_rv_plic_component').
-        if template_core_name.startswith(self.ip_template.name):
+        if "module_instance_name" in self.ip_config.param_values:
+            idx = len(self.ip_config.param_values["module_instance_name"])
+            template_core_name = template_core_name[idx:]
```

Comment:
I think this probably also needs an assertion to check that `template_core_name` starts with `self.ip_config.param_values["module_instance_name"]`?

Reply:
It should be the case from the template, but adding a check here is definitely useful.